### PR TITLE
feat(zscript): add `RepairShield` function to npc

### PR DESCRIPTION
**Pull request by Jambu2024**
npc->RepairShield() compiles find.
but it doesn't appear to be doing anything...

### CODE REVIEW

File: src/zasm/table.cpp [hunk 1]
```diff
@@ -240,6 +240,7 @@ static constexpr script_command command_list[]=
 	{ "DRAWLAYERR", DRAWLAYERR, 0, {}, 0, 0 },
 	{ "DRAWSCREENR", DRAWSCREENR, 0, {}, 0, 0 },
 	{ "BREAKSHIELD", BREAKSHIELD, 1, { REG_R }, 0, 0 },
+	{ "REPAIRSHIELD", REPAIRSHIELD, 1, { REG_R }, 0, 0 },
```
Comment (EmilyV99) [inline src/zasm/table.cpp]: pretty sure you need to keep this in order (so only add at the end of the list)?

File: resources/include/bindings/npc.zh [hunk 2]
```diff
@@ -651,6 +651,11 @@ class npc {
 	//   BREAKSHIELD D2
```
Comment (EmilyV99) [inline resources/include/bindings/npc.zh:651]: Something looks wrong to me here, in the already-existing code... why is REFNPC not being used here at all? @connorjclark

Comment (connorjclark) [reply resources/include/bindings/npc.zh:651]: It's actually correct. The reason it doesn't set REFNPC is that is how the instructions were written - see the releases/2.55 npc symbols, and see how `do_breakshield` doesn't use npcref but instead uses sarg1. This `@zasm` code was not handwritten but came directly from those old symbol cpp files.

Comment (EmilyV99) [reply resources/include/bindings/npc.zh:651]: Ah, fun. Then @Jambu2024 I wouldn't be copying from that for your new code, I'd copy from something that uses the REFNPC register.

Comment (Jambu2024) [reply resources/include/bindings/npc.zh:651]: redid everything

**Comment by Jambu2024** [comment]
i tried updating the zasm snapshot but test_zscript. is returning (could not parse)

**Comment by connorjclark** [comment]
> i tried updating the zasm snapshot but test_zscript. is returning (could not parse)

If you share the whole output I could maybe diagnose.

It worked locally for me - I tried pushing to your branch but do not have access.

> error: Authentication error: Authentication required: You must have push access to verify locks
> error: failed to push some refs to 'https://github.com/Jambu2024/ZQuestClassic.git'

The patch:

```
commit a5fedbf2903f3bda2be82b5ab0bde495ef59bb6a
Author: Connor Clark <cjamcl@gmail.com>
Date:   Mon Nov 11 21:03:36 2024 -0800

    update snapshot

diff --git a/tests/scripts/misc/metadata_expected.txt b/tests/scripts/misc/metadata_expected.txt
index 643886dbb..0e89c2776 100644
--- a/tests/scripts/misc/metadata_expected.txt
+++ b/tests/scripts/misc/metadata_expected.txt
@@ -884,7 +884,7 @@ stdout:
         },
         "doc": "THIS COMMENT USED #1"
       },
-      "2403": {
+      "2404": {
         "loc": {
           "range": {
             "start": {
@@ -962,7 +962,7 @@ stdout:
         },
         "doc": "x marks the spot\nstill line one still line one still line one still line one still line one\n\nline two\nline two still\n\nline three\n\n- list item 1\n- list item 2\n- list item 3"
       },
-      "2405": {
+      "2406": {
         "loc": {
           "range": {
             "start": {
@@ -994,7 +994,7 @@ stdout:
         },
         "doc": "A car, duh"
       },
-      "2409": {
+      "2410": {
         "loc": {
           "range": {
             "start": {
@@ -1168,7 +1168,7 @@ stdout:
           "uri": "URI"
         }
       },
-      "3143": {
+      "3144": {
         "loc": {
           "range": {
             "start": {
@@ -1324,7 +1324,7 @@ stdout:
         },
         "doc": "How fast it goes. See [`the vroom method`](command:zscript.openLink?%7B%22file%22%3A%22metadata.zs%22%2C%22position%22%3A%7B%22start%22%3A%7B%22line%22%3A31%2C%22character%22%3A9%7D%2C%22end%22%3A%7B%22line%22%3A31%2C%22character%22%3A14%7D%7D%7D) or [`speed`](command:zscript.openLink?%7B%22file%22%3A%22metadata.zs%22%2C%22position%22%3A%7B%22start%22%3A%7B%22line%22%3A23%2C%22character%22%3A8%7D%2C%22end%22%3A%7B%22line%22%3A23%2C%22character%22%3A13%7D%7D%7D) or [`x`](command:zscript.openLink?%7B%22file%22%3A%22metadata.zs%22%2C%22position%22%3A%7B%22start%22%3A%7B%22line%22%3A16%2C%22character%22%3A4%7D%2C%22end%22%3A%7B%22line%22%3A16%2C%22character%22%3A5%7D%7D%7D) or [`Waitframe()`](command:zscript.openLink?%7B%22file%22%3A%22global.zh%22%2C%22position%22%3A%7B%22start%22%3A%7B%22line%22%3A27%2C%22character%22%3A14%7D%2C%22end%22%3A%7B%22line%22%3A27%2C%22character%22%3A23%7D%7D%7D) or [`bitmap`](command:zscript.openLink?%7B%22file%22%3A%22bitmap.zh%22%2C%22position%22%3A%7B%22start%22%3A%7B%22line%22%3A3%2C%22character%22%3A6%7D%2C%22end%22%3A%7B%22line%22%3A3%2C%22character%22%3A12%7D%7D%7D) or [`bitmap::Width`](command:zscript.openLink?%7B%22file%22%3A%22bitmap.zh%22%2C%22position%22%3A%7B%22start%22%3A%7B%22line%22%3A13%2C%22character%22%3A20%7D%2C%22end%22%3A%7B%22line%22%3A13%2C%22character%22%3A25%7D%7D%7D)"
       },
-      "2407": {
+      "2408": {
         "loc": {
           "range": {
             "start": {
@@ -1340,7 +1340,7 @@ stdout:
         },
         "doc": "Go fast"
       },
-      "4919": {
+      "4920": {
         "loc": {
           "range": {
             "start": {
@@ -1379,7 +1379,7 @@ stdout:
           "character": 18,
           "length": 1
         },
-        "symbol": "2403"
+        "symbol": "2404"
       },
       {
         "loc": {
@@ -1427,7 +1427,7 @@ stdout:
           "character": 15,
           "length": 2
         },
-        "symbol": "2405"
+        "symbol": "2406"
       },
       {
         "loc": {
@@ -1451,7 +1451,7 @@ stdout:
           "character": 21,
           "length": 3
         },
-        "symbol": "2409"
+        "symbol": "2410"
       },
       {
         "loc": {
@@ -1547,7 +1547,7 @@ stdout:
           "character": 21,
           "length": 13
         },
-        "symbol": "3143"
+        "symbol": "3144"
       },
       {
         "loc": {
@@ -1563,7 +1563,7 @@ stdout:
           "character": 21,
           "length": 3
         },
-        "symbol": "2409"
+        "symbol": "2410"
       },
       {
         "loc": {
@@ -1699,7 +1699,7 @@ stdout:
           "character": 8,
           "length": 5
         },
-        "symbol": "2407"
+        "symbol": "2408"
       },
       {
         "loc": {
@@ -1715,7 +1715,7 @@ stdout:
           "character": 22,
           "length": 5
         },
-        "symbol": "4919"
+        "symbol": "4920"
       }
     ]
   }
```

File: src/zasm/table.cpp (hunk 1 above)
Comment (connorjclark) [reply src/zasm/table.cpp]: Order does not matter - the `sort_script_commands` constexpr function allows for that.

**Pull request merged**